src/find-options/FindOptionsUtils.ts

```diff
@@ -87,13 +90,22 @@ export class FindOptionsUtils {
     /**
      * Applies give find options to the given query builder.
      */
-    static applyOptionsToQueryBuilder<T>(qb: SelectQueryBuilder<T>, options: FindOneOptions<T>|FindManyOptions<T>|undefined): SelectQueryBuilder<T> {
+    static applyOptionsToQueryBuilder<T>(qb: SelectQueryBuilder<T>, options: FindOneOptions<T>|FindManyOptions<T>|undefined, metadata: EntityMetadata): SelectQueryBuilder<T> {
 
         // if options are not set then simply return query builder. This is made for simplicity of usage.
         if (!options || !this.isFindOneOptions(options))
             return qb;
 
         // apply all options from FindOptions
+        if (options.select) {
+            qb.select(options.select.map(selection => qb.alias + "." + selection));
+
+            metadata.primaryColumns
```

Comment:
its a wrong place for this code. This code should be part of query builder. Most probably in `createSelectExpression` method where you can check if primary columns are not selected and select them just like what you do here. Also add test which do partial selection using query builder only (without using find methods - because find methods are just delegation to query builder)